### DISCUSSION TITLE
Helix release cycle

### DISCUSSION THREAD

**Discussion by invakid404** [helix-editor/helix#4803] [Q&A] [answered]
I noticed there have been almost 500 commits since the last release, which got me wondering what Helix's release cycle is. There are some things I'd like to try that have been implemented since the latest release, yet I don't feel like building from master...

**Comment by David-Else** [comment]
I think it is roughly every three months, a new `22.11` release is due soon https://github.com/helix-editor/helix/pulls?q=is%3Aopen+is%3Apr+milestone%3A22.11

I am not sure why this cadence was chosen as it seems too long a wait for software that is developing so fast.

- As most people are on the official release the issues are polluted by people reporting bugs that are already fixed months ago
- New features are only being tested by the people who are compiling from master regularly, so the releases have a lot of new code that has not really been widely tested
- The majority of users are waiting 3 months to enjoy all the new features, and there are a lot of amazing new features being developed

I think the argument against more regular release is that it is so easy to compile using cargo there is no reason not to, and each official release requires package maintainers to do work so don't overtax them? This seems logical, but I think in reality people don't want to build from master unless they have to, so it might be a bit of bias that Rust developers have thinking everyone is happy to compile from source.

**Reply by invakid404** [reply]
I personally don't mind compiling from source, I just generally dislike bleeding-edge stuff. And as a downstream maintainer for Funtoo Linux, I can easily arrange a Helix package that builds from master, it'd just leave a bad aftertaste in my mouth.

That being said, there are some projects that explicitly state that it's fine for downstream maintainers to package master (such as [mpv](https://github.com/mpv-player/mpv#release-cycle)) so really I was wondering if Helix is one of these projects. The readme kinda suggests so with building from source being recommended, but the occasional CI failures on master don't give me much confidence to do so as a downstream maintainer, hence why I'm raising this question.

**Comment by the-mikedavis** [comment, marked as answer]
See https://github.com/helix-editor/helix/wiki/FAQ#when-will-the-next-release-be

Using release versions is a good choice for package managers - we focus on stability for a few weeks leading up to the release to iron out bugs on master. There are packages that follow master like [`helix-git` in the AUR](https://aur.archlinux.org/packages/helix-git) though and those are also fine. master is generally pretty stable but sometimes we see new panics or unintended behavior changes be introduced, and these are usually taken care of by release time.

**Reply by invakid404** [reply]
Ah, I missed that in the wiki, I went straight to issues/discussions and saw nothing, so I thought I'd ask. Thanks!
